### PR TITLE
Fixing bug in Actors where it cannot deactivate actors with UUID as id.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fixing issue where Actor with Id as UUID cannot be deactivated because parsing uses "-" as separator.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #895

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Unit tests passing
* [?] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
+const actorTypeIDSeparator = "$"
```
Comment (amanbha) [inline pkg/actors/actors.go]: an actor id can contain $ as well. Use some more complex actor separator something like __delimit__

**Comment by amanbha** [comment]
> Should `constructActorStateKey` be changed as well?

Currently, we dont split the state keys & reminder keys, we create the keys and use them to fetch/update their values. But, yes, it would be good to change them as well. it would make the code immune to future changes if we ever split based on delimiter in code.

File: pkg/actors/actors.go (hunk 1 above)
Comment (artursouza) [reply pkg/actors/actors.go]: Let's discuss this since we might need to use some sort of encoding before concatenation.

Comment (LMWF) [reply pkg/actors/actors.go]: Do we document what characters can be in an actor id since it's simply a string?

Comment (artursouza) [reply pkg/actors/actors.go]: The question is whether to limit to to support extra character via encoding.

Comment (artursouza) [reply pkg/actors/actors.go]: Done. Please, check the latest version. I would like to wait for the new E2E test for actors to me merged first: #902

File: pkg/actors/actors.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
+const actorTypeIDSeparator = "__DELIM__"
```
[the remark this answers is not in the document]
Comment (amanbha) [reply pkg/actors/actors.go]: Corresponding changes are needed for state management with Dapr (outside of actor) as discussed.
The documentation for querying keys also needs to be updated: https://github.com/dapr/docs/blob/master/howto/query-state-store/query-redis-store.md
https://github.com/dapr/docs/blob/master/howto/query-state-store/query-cosmosdb-store.md

[the remark this answers is not in the document]
Comment (artursouza) [reply pkg/actors/actors.go]: Makes sense. I only found documentation on this. I looked at the Cassandra and Redis components and they don't do concatenation or split - I assume this is handled at the runtime. Is there any other code I need to fix for delim change? I will work on the documentation changes before merging this change.

[the remark this answers is not in the document]
Comment (artursouza) [reply pkg/actors/actors.go]: Done. Please, check the new files I changed in this PR and also the PR on docs: https://github.com/dapr/docs/pull/279

I am still not comfortable with the runtime code, so please advise.

**Pull request merged**